test/python/test_linear_attention_gdn.py

```
@@ -0,0 +1,153 @@
+        scale = 1.0 / math.sqrt(K)
+        o, _ = gated_delta_net(q, k, v, g, beta, scale=scale, output_final_state=False)
+        o_ref, _ = _reference_recurrent(q, k, v, g, beta, scale)
+        assert _rms_ratio(o, o_ref) < 2e-2
+
+    @pytest.mark.parametrize("B,T,H,K,V", [
+        (1, 128, 1, 64, 64),
+        (2, 128, 2, 64, 64),
+    ])
+    def test_backward_runs(self, B, T, H, K, V):
+        q, k, v, g, beta = _rand_inputs(B, T, H, K, V)
+        q.requires_grad_(True); k.requires_grad_(True); v.requires_grad_(True)
```

Ruff (0.15.20)

[error] Multiple statements on one line (semicolon)

(E702)